src/scanner.re

```diff
@@ -220,11 +220,15 @@ int Scanner::scan(YYSTYPE* yylval)
   INITIAL                = CONSTITUENT | SPECIAL_INITIAL | INLINE_HEX_ESCAPE | "-"; /* we allow '-' identifier for shell mode */
   SUBSEQUENT             = INITIAL | DIGIT | [\+\-\.@]; /* todo: Add Unicode category Nd, Mc and Me */
   PECULIAR_IDENTIFIER    = [\+\-] | "..." | ("->" (SUBSEQUENT)*) | "@"; /* "@" is not R6RS match.scm required it. */
-  IDENTIFIER             = (INITIAL (SUBSEQUENT)*) | PECULIAR_IDENTIFIER;
   SHEBANG                = "#!" [^\n\X0000]* (LINE_ENDING | EOS);
   COMMENT                = (";"[^\n\X0000]* (LINE_ENDING | EOS)) | ("#!" [a-zA-Z0-9/\_\.\-]+);
   DEFINING_SHARED        = "#" DIGIT+ "=";
   DEFINED_SHARED         = "#" DIGIT+ "#";
+  MNEMONIC_ESCAPE        = ('\\' [abtnr]);
+  /* R7RS doesn't have " here but r7rs-tests.scm requires it */
```

Comment:
It's R7RS errata. From: https://github.com/johnwcowan/r7rs-work/blob/master/R7RSSmallErrata.md

>  This makes the same escape sequences valid in both strings and symbols.

So we can assume "if it's valid for a string, it's always valid for a symbol."

Reply:
Thanks for the link. Makes sense. I added \\\\ and \\\" as symbol element as well.
Per conversation above I'll merge this to master once the CI becomes green.